experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -93,6 +93,11 @@ export interface SharedTreeBranchEvents<TEditor extends ChangeFamilyEditor, TCha
 	 */
 	change(change: SharedTreeBranchChange<TChange>): void;
 
+	/**
+	 * Fired when an undoable change is made to this branch.
+	 */
+	undoable(type: UndoRedoManagerCommitType): void;
```

Comment:
I think that it might be best to merge this with the `change` event. It seems reasonable that somebody might want to listen for e.g. a redo and then also look at the commit or the change or something; it'd be nice if they could do it all in one event. You can simply add `type` as a second parameter to the `change` event and another parameter to `emitAndRebaseAnchors`.

Reply:
I ended up deciding not to merge these and renamed it to revertible because there are some situations where we want to send revertible but it doesn't seem like we want to send change. Namely, during merge, we send a revertible for every new commit merged onto the tree while we only send a single change event.

Reply:
That makes sense, sounds good to me. Merge and rebase always mess everything up, don't they?